tiatoolbox/models/abc.py

```diff
@@ -65,7 +65,7 @@ def __init__(self):
 
     @abstractmethod
     # pylint: disable=W0221
-    # shut up pylint because this is generic abc, else it will keep complain
+    # shut up pylint because this is generic abc, else it will keep complaining
     def forward(self, *args, **kwargs):
         """Torch method, this contains logic for using layers defined in init."""
         ...
```

Comment:
why not `raise NotImplementedError`

Reply:
same for line 86

Reply:
It triggers class checking earlier.

https://stackoverflow.com/a/44316506

Reply:
but you can add a test using `with pytest.raises(NotImplementedError)`

Reply:
It triggers TypeError, and you can see the test for this here https://github.com/TIA-Lab/tiatoolbox/blob/325b9d15a86c16d293407143cb377c12a87d11eb/tests/test_models.py#L550-L559. This is better because of this

>An advantage of this over NotImplementedError is that you get an explicit Exception at instantiation time, not at method call time.